include/NeoFOAM/fields/domainField.hpp

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 NeoFOAM authors
+#pragma once
+
+#include <Kokkos_Core.hpp>
+#include <iostream>
+#include "NeoFOAM/primitives/label.hpp"
+#include "NeoFOAM/primitives/scalar.hpp"
+#include "boundaryFields.hpp"
+
+#include "NeoFOAM/core/executor/executor.hpp"
+
+namespace NeoFOAM
+{
+
+template<typename T>
+class DomainField
+{
+    /**
+     * @class domainField
+     * @brief Represents the domain fields for a computational domain.
+     *
+     * The domainField class stores the internal fields and boundary information for a computational domain.
+     * It provides access to the computed values, reference values, value fractions, reference gradients, boundary types,
+     * offsets, and the number of boundaries and boundary faces.
+     */
```

Comment:
doc string before class
```suggestion
/**
* @class domainField
* @brief Represents the domain fields for a computational domain.
*
* The domainField class stores the internal fields and boundary information for a computational domain.
* It provides access to the computed values, reference values, value fractions, reference gradients, boundary types,
* offsets, and the number of boundaries and boundary faces.
*/
template<typename T>
class DomainField
{
```